### DISCUSSION TITLE
Index media directories

### DISCUSSION THREAD

**Discussion by IcyAlmond** [Syncplay/syncplay#547] [Archive]
Is it possible to have syncplay index the media directories beforehand so that it doesn't stall on larger directories or network directories?

I'm not sure how it works, but other programs like https://github.com/cboxdoerfer/fsearch index directories before hand and save it to disk and searches very fast afterwards. (and the index time is quite short too given how many files it has to scan)

I think this would be especially helpful on network mounts if syncplay could index and save the indexed data to the disk as syncplay would know where files are instead of having to scan slow network drives every time it needs to find something.

**Comment by Et0h** [comment]
At present we advise that people don't use Syncplay with large directories or network directories, and instead create a local folder or sub-folder for the files which you want to be accessible to Syncplay. As the current system works for local non-large directories there hasn't been anyone who has volunteered to improve how indexing works (maybe because reorganising their file system is quicker than changing how Syncplay works and testing it on all operating systems and use cases).

If someone wanted to improve Syncplay handling of directories then the priority would probably be to change the behaviour to reduce how often Syncplay re-scans a directory and instead rely on something like https://pypi.org/project/watchdog/ to be notified when a new file is added to a folder.

Given that these are media folders it is important that whatever system is used detects changes which occur when Syncplay is off and when Syncplay is on.

There might be privacy implications for cache systems and so any implementation would need careful consideration.

**Comment by Et0h** [comment]
This discussion is being locked because nobody has volunteered to take this matter forwards. If someone wishes to develop this feature then please let us know.
